### PR TITLE
[JAVA] [JDBC] Change IntervalAccessor getString format

### PR DESCRIPTION
**Pull request by escobargabriel**
Change `IntervalAccessor#getString` to format periods similar to Oracle's representation of the "INTERVAL * to *" data type instead of the representation given by the Interval vector's StringBuilder.

### CODE REVIEW

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/calendar/ArrowFlightJdbcIntervalVectorAccessorTest.java [hunk 1]
```diff
@@ -134,10 +136,13 @@ public void testShouldGetObjectReturnNull() throws Exception {
...
-    if (vector instanceof IntervalDayVector) {
-      return ((IntervalDayVector) vector).getAsStringBuilder(index).toString();
+    String object = getExpectedObject(vector, index).toString();
+    if (object == null) {
+      return null;
+    } else if (vector instanceof IntervalDayVector) {
+      return formatIntervalDay(org.joda.time.Period.parse(object));
     } else if (vector instanceof IntervalYearVector) {
-      return ((IntervalYearVector) vector).getAsStringBuilder(index).toString();
+      return formatIntervalYear(org.joda.time.Period.parse(object));
```
Comment (vfraga) [inline java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/calendar/ArrowFlightJdbcIntervalVectorAccessorTest.java:145]: Let's make `parse` a static import here too.

File: java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/accessor/ArrowFlightJdbcAccessor.java [hunk 2]
```diff
@@ -250,6 +251,57 @@ public Reader getNCharacterStream() throws SQLException {
...
+    millis = millis % DateUtility.secondsToMillis;
+
+    return String.format("%c%03d %02d:%02d:%02d.%03d", neg ? '-' : '+', days, hours, minutes, seconds, millis);
+  }
+
+  public static int millisFromPeriod(org.joda.time.Period period) {
+    return period.getHours() * DateUtility.hoursToMillis + period.getMinutes() * DateUtility.minutesToMillis +
+        period.getSeconds() * DateUtility.secondsToMillis + period.getMillis();
+  }
+
```
Comment (vfraga) [inline java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/accessor/ArrowFlightJdbcAccessor.java:304]: Can these methods be in an utility class? Let's avoid static methods that can potentially be used elsewhere inside the accessors.

File: java/flight/flight-integration-tests/pom.xml [hunk 3]
```diff
@@ -59,6 +59,11 @@
             <groupId>org.slf4j</groupId>
             <artifactId>slf4j-api</artifactId>
         </dependency>
+        <dependency>
+            <groupId>joda-time</groupId>
+            <artifactId>joda-time</artifactId>
+            <version>2.10.14</version>
+        </dependency>
```
Comment (vfraga) [inline java/flight/flight-integration-tests/pom.xml:66]: Why is this needed in the integration tests module?

**Review by vfraga — CHANGES_REQUESTED** [review]

File: java/flight/flight-integration-tests/pom.xml (hunk 3 above)
Comment (escobargabriel) [reply java/flight/flight-integration-tests/pom.xml:66]: Removed.

File: java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/accessor/impl/calendar/ArrowFlightJdbcIntervalVectorAccessor.java [hunk 4]
```diff
@@ -91,15 +92,20 @@ public Object getObject() {
...
       return null;
     }
-
-    return stringBuilder.toString();
+    if (vector instanceof IntervalDayVector) {
+      return formatIntervalDay(org.joda.time.Period.parse(object.toString()));
+    } else if (vector instanceof IntervalYearVector) {
+      return formatIntervalYear(org.joda.time.Period.parse(object.toString()));
+    } else {
+      throw new SQLException("Invalid Interval vector instance");
```
[the remark this answers is not in the document]
Comment (escobargabriel) [reply java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/accessor/impl/calendar/ArrowFlightJdbcIntervalVectorAccessor.java:108]: Done.

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/calendar/ArrowFlightJdbcIntervalVectorAccessorTest.java (hunk 1 above)
Comment (escobargabriel) [reply java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/calendar/ArrowFlightJdbcIntervalVectorAccessorTest.java:145]: Done.

File: java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/accessor/ArrowFlightJdbcAccessor.java (hunk 2 above)
Comment (escobargabriel) [reply java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/accessor/ArrowFlightJdbcAccessor.java:304]: Done.

File: java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/utils/IntervalDayYearUtils.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+package org.apache.arrow.driver.jdbc.utils;
+
+import org.apache.arrow.vector.util.DateUtility;
+
+/**
+ * Utility class to format periods similar to Oracle's representation
+ * of "INTERVAL * to *" data type.
+ */
+public class IntervalDayYearUtils {
```
Comment (vfraga) [inline java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/utils/IntervalDayYearUtils.java:26]: Maybe "IntervalStringUtils"?

File: java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/utils/IntervalDayYearUtils.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+  /**
+   * Constructor Method of class.
+   */
+  private IntervalDayYearUtils( ) {}
+
+  /**
+   * Formats a period similar to Oracle INTERVAL YEAR TO MONTH data type<br>.
+   * For example, the string "+21-02" defines an interval of 21 years and 2 months.
+   */
+  public static String formatIntervalYear(final org.joda.time.Period p) {
```
Comment (vfraga) [inline java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/utils/IntervalDayYearUtils.java:37]: This can be just `Period`

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/calendar/ArrowFlightJdbcIntervalVectorAccessorTest.java [hunk 7]
```diff
@@ -134,14 +137,27 @@ public void testShouldGetObjectReturnNull() throws Exception {
...
     } else if (vector instanceof IntervalYearVector) {
-      return ((IntervalYearVector) vector).getAsStringBuilder(index).toString();
+      return formatIntervalYear(parse(object));
     }
     return null;
   }
 
+  @Test
+  public void testShouldGetNegativeIntervalYear( ) {
+    formatIntervalYear(org.joda.time.Period.parse("P-1Y-1M"));
```
Comment (vfraga) [inline java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/calendar/ArrowFlightJdbcIntervalVectorAccessorTest.java:153]: This can be just "parse"

**Review by vfraga — CHANGES_REQUESTED** [review]

File: java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/utils/IntervalDayYearUtils.java (hunk 6 above)
Comment (escobargabriel) [reply java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/utils/IntervalDayYearUtils.java:37]: Done.

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/calendar/ArrowFlightJdbcIntervalVectorAccessorTest.java (hunk 7 above)
Comment (escobargabriel) [reply java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/calendar/ArrowFlightJdbcIntervalVectorAccessorTest.java:153]: Done.

Comment (rafael-telles) [reply java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/calendar/ArrowFlightJdbcIntervalVectorAccessorTest.java:153]: Missing asserts?

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/calendar/ArrowFlightJdbcIntervalVectorAccessorTest.java [hunk 8]
```diff
@@ -134,14 +137,27 @@ public void testShouldGetObjectReturnNull() throws Exception {
...
+  }
+
+  @Test
+  public void testShouldGetNegativeIntervalDay( ) {
+    formatIntervalDay(org.joda.time.Period.parse("PT-1H"));
```
Comment (rafael-telles) [inline java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/calendar/ArrowFlightJdbcIntervalVectorAccessorTest.java:158]: Missing asserts?

**Review by rafael-telles — CHANGES_REQUESTED** [review]

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/calendar/ArrowFlightJdbcIntervalVectorAccessorTest.java (hunk 7 above)
Comment (escobargabriel) [reply java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/calendar/ArrowFlightJdbcIntervalVectorAccessorTest.java]: Done.

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/calendar/ArrowFlightJdbcIntervalVectorAccessorTest.java (hunk 8 above)
Comment (escobargabriel) [reply java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/calendar/ArrowFlightJdbcIntervalVectorAccessorTest.java]: Done.

File: java/flight/flight-jdbc-driver/pom.xml [hunk 9]
```diff
@@ -143,6 +143,12 @@
             <artifactId>bcpkix-jdk15on</artifactId>
             <version>1.61</version>
         </dependency>
+
+        <dependency>
+            <groupId>joda-time</groupId>
```
Comment (jduo) [inline java/flight/flight-jdbc-driver/pom.xml:148]: Is JodaTime really necessary? I thought most of its functionality is part of JDK 8.

Comment (escobargabriel) [reply java/flight/flight-jdbc-driver/pom.xml:148]: The joda-time lib helps simplify parsing java.time.Duration and java.time.Period in fewer methods. It's also what is used in Dremio, so it's easier to ensure it has the same value.

File: java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/accessor/ArrowFlightJdbcAccessor.java [hunk 10]
```diff
@@ -38,6 +38,7 @@
 import java.util.Map;
 import java.util.function.IntSupplier;
 
+
```
Comment (jduo) [inline java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/accessor/ArrowFlightJdbcAccessor.java:41]: Unnecessary

Comment (escobargabriel) [reply java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/accessor/ArrowFlightJdbcAccessor.java:41]: Done.

File: java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/utils/IntervalStringUtils.java [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+package org.apache.arrow.driver.jdbc.utils;
+
+import org.apache.arrow.vector.util.DateUtility;
+import org.joda.time.Period;
+
+/**
+ * Utility class to format periods similar to Oracle's representation
+ * of "INTERVAL * to *" data type.
+ */
+public class IntervalStringUtils {
```
[the remark this answers is not in the document]
Comment (escobargabriel) [reply java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/utils/IntervalStringUtils.java:27]: Done.
